### PR TITLE
Feature/insights alert rules

### PR DESCRIPTION
**Pull request by mytev**
- Add application insights alert rule support
- Fix WebTest `_ResourceId` property

### CODE REVIEW

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/Add-ArmApplicationInsightsAlertRuleWebhookAction.ps1 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+function Add-ArmApplicationInsightsAlertRuleWebhookAction {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType("ApplicationInsightsAlertRule")]
+    Param(
+        [PSTypeName("ApplicationInsightsAlertRule")]
+        [Parameter(Mandatory, ValueFromPipeline)]
+        $ApplicationInsightsAlertRule,
+        [Parameter(Mandatory)]
+        [string]
+        $ServiceUri,
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/Add-ArmApplicationInsightsAlertRuleWebhookAction.ps1]: Is this the webhook url? If so, we should rename to Webhook

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "Minimum",
+            "Maximum",
+            "Total",
+            "Last")]
+        [string]
+        $TimeAggregation,
+        [int]
+        $FailedLocationCount,
+        [string]
+        $WindowSize = "PT5M",
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1]: There a ValidSet for this

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1 [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        [ValidateSet(
+            "RuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.LocationThresholdRuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.ManagementEventRuleCondition")]
+        [string]
+        $ConditionOdataType,
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1]: We should rename this to Condition and have a simplified ValidSet and then in the function map to the correct long OdataType

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1 [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            _ResourceId = $Name | New-ArmFunctionResourceId -ResourceType 'microsoft.insights/alertrules'
+            PSTypeName  = "ApplicationInsightsAlertRule"
+            type        = 'microsoft.insights/alertrules'
+            name        = $Name
+            apiVersion  = $ApiVersion
+            location    = $ApplicationInsights.location
+            properties  = @{
+                name        = $Name
+                description = $Description
+                isEnabled   = -not $Disable.ToBool()
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1]: Disabled*

**Review by mbernard — CHANGES_REQUESTED** [review]

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/Add-ArmApplicationInsightsAlertRuleWebhookAction.ps1 (hunk 1 above)
Comment (mytev) [reply src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/Add-ArmApplicationInsightsAlertRuleWebhookAction.ps1]: Yes it is. Renamed to `Webhook`

**Review by mbernard — APPROVED** [review]

**Pull request merged**